src/modules/ekf2/ekf2_main.cpp

```diff
@@ -570,14 +576,22 @@ void Ekf2::run()
 			orb_copy(ORB_ID(optical_flow), optical_flow_sub, &optical_flow);
 		}
 
-		orb_check(range_finder_sub, &range_finder_updated);
+		for (unsigned i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
 
-		if (range_finder_updated) {
-			orb_copy(ORB_ID(distance_sensor), range_finder_sub, &range_finder);
+			orb_check(range_finder_subs[i], &range_finder_updated);
+
+			if (range_finder_updated) {
 
-			if (range_finder.min_distance >= range_finder.current_distance
-			    || range_finder.max_distance <= range_finder.current_distance) {
-				range_finder_updated = false;
+				orb_copy(ORB_ID(distance_sensor), range_finder_subs[i], &range_finder);
+
+				if (range_finder.orientation != distance_sensor_s::ROTATION_DOWNWARD_FACING
+				    || range_finder.min_distance >= range_finder.current_distance
+				    || range_finder.max_distance <= range_finder.current_distance) {
+					range_finder_updated = false;
+
+				} else {
+					break; // already the correct one -> break for loop
+				}
```

Comment:
This is definitely an improvement. Two comments:
- only the first valid range instance is used, others are discarded, right? If yes can you add a comment?
- ekf2 buffers range samples (see here https://github.com/PX4/ecl/blob/4411c7fd1c0fcf76c6b42b80720683d22a2c42d5/EKF/estimator_interface.cpp#L251). With multi instances the buffer might contain data from different sensors, I wonder if that can have unwanted side effects. Maybe @priseborough can comment on this?

Reply:
Multiple downwards looking sensors should be avoided.

The EKF parameters contain the body frame position offset for a single range finder. Multiple downwards looking sensors would need to be co-located, but then there is the risk of mutual interference.

Our longer term goal for all sensor offsets should be to move their definition out of the EKF2 parameters and into the sensor module so that they are published with the sensor data.

Reply:
Ok! In the meantime @ChristophTobler modified this part to make sure that the ekf is updated with a single downward looking sensor. So that should not be a problem.

Btw great idea to pack all required info in the sensor topics, i'm looking forward to that!